File: src/org/jgll/util/ToDotWithoutIntermediateNodes.java
package org.jgll.util;

import org.jgll.sppf.NonterminalSymbolNode;
import org.jgll.sppf.PackedNode;
import org.jgll.traversal.SPPFVisitor;
import org.jgll.traversal.SPPFVisitorUtil;

/**
 * Creates a Graphviz's dot format representation of an SPPF node.
 * 
 * @author Ali Afroozeh
 * 
 * @see SPPFVisitor
 */
public class ToDotWithoutIntermediateNodes extends ToDot {
	
	@Override
	public void visit(PackedNode node, StringBuilder sb) {
		SPPFVisitorUtil.removeIntermediateNode(node);
		super.visit(node, sb);
	}
	
	@Override
	public void visit(NonterminalSymbolNode node, StringBuilder sb) {
		SPPFVisitorUtil.removeIntermediateNode(node);
		super.visit(node, sb);
	}
}

File: src-scala/org/jgll_staged/util/dot/ToDotWithoutIntermediateNodes.scala
package org.jgll_staged.util.dot

import org.jgll_staged.sppf.NonterminalSymbolNode
import org.jgll_staged.sppf.PackedNode
import org.jgll_staged.traversal.SPPFVisitor
import org.jgll_staged.traversal.SPPFVisitorUtil
//remove if not needed
import scala.collection.JavaConversions._

class ToDotWithoutIntermediateNodes extends SPPFToDot {

  override def visit(node: PackedNode) {
    SPPFVisitorUtil.removeIntermediateNode(node)
    super.visit(node)
  }

  override def visit(node: NonterminalSymbolNode) {
    SPPFVisitorUtil.removeIntermediateNode(node)
    super.visit(node)
  }
}
